modAionImpl/src/org/aion/zero/impl/pendingState/AionPendingStateImpl.java

```diff
@@ -280,140 +287,102 @@ private void addTransactionsFromCacheOrBackup(List<AionTransaction> transactions
         List<TxResponse> txResponses = new ArrayList<>();
 
         for (AionTransaction tx : transactions) {
-            BigInteger txNonce = tx.getNonceBI();
-            BigInteger bestPSNonce = bestPendingStateNonce(tx.getSenderAddress());
-            AionAddress txFrom = tx.getSenderAddress();
-
-            int cmp = txNonce.compareTo(bestPSNonce);
+            BigInteger bestPendingStateNonce = bestPendingStateNonce(tx.getSenderAddress());
+            int cmp = tx.getNonceBI().compareTo(bestPendingStateNonce);
 
-            // This case happens when we have already received a tx with a larger nonce
-            // from the address txFrom
-            if (cmp > 0) {
-                if (isInTxCache(txFrom, txNonce)) {
-                    txResponses.add(TxResponse.ALREADY_CACHED);
-                } else {
-                    newLargeNonceTx.add(tx);
-                    addToTxCache(tx);
-
-                    if (poolBackUpEnable) {
-                        backupPendingCacheAdd.put(tx.getTransactionHash(), tx.getEncoded());
-                    }
-
-                    if (LOGGER_TX.isTraceEnabled()) {
-                        LOGGER_TX.trace(
-                                "addPendingTransactions addToCache due to largeNonce: from = {}, nonce = {}",
-                                txFrom,
-                                txNonce);
-                    }
+            if (pendingTxCache.isInCache(tx.getSenderAddress(), tx.getNonceBI())) {
+                txResponses.add(TxResponse.ALREADY_CACHED);
+            } else {
+                // This case happens when we have already received a tx with a larger nonce
+                // from the address txFrom
+                if (cmp > 0) {
+                    TxResponse response = addTransactionToCache(tx, false);
```

Comment:
Could this code be split into a smaller method? It would allow easier testing and make the functionality easier to read. The same comment applies to the other two cases for the `cmp` values.